src/mgmt/manager/project/api/ports.py

```diff
@@ -46,48 +46,72 @@ def ports_update_eps_config():
 
 @ports_blueprint.route('/ports', methods=['GET', 'POST'])
 def all_ports():
+    status_code = 200
     if request.method == 'POST':
         portList = request.get_json()
         amount_of_ports = len(portList)
         logger.debug(f'Start to make {amount_of_ports} ports.')
         start_time = time.time()
+        ports_to_add = []
+        eps_to_add = []
+        hosts_to_add = []
+        host_ip_node_lookup_set = set()
         for post_data in portList:
-            port = Port(
-                port_id = post_data.get('port_id'),
-                mac_port = post_data.get('mac_port'),
-                vpc_id = post_data.get('vpc_id'))
-
-            host = Host.query.filter_by(ip_node=post_data.get('ip_node')).first()
-            if host is None:
-                host = Host(
-                    mac_node = post_data.get('mac_node'),
-                    ip_node = post_data.get('ip_node'))
-                host.host_id=str(uuid.uuid4())
-                db.session.add(host)
-            port.host_id = host.host_id
+            port = {
+                'port_id': post_data.get('port_id'),
+                'mac_port': post_data.get('mac_port'),
+                'vpc_id': post_data.get('vpc_id'),
+                'host_id': '',
+                'eps': []
+            }
 
+            ip_node_to_find = post_data.get('ip_node')
+            host = Host.query.filter_by(ip_node=ip_node_to_find).first()
+            if (host is None) and (ip_node_to_find not in host_ip_node_lookup_set):
+                host_to_add = {'mac_node': post_data.get('mac_node'),
+                    'ip_node' : ip_node_to_find,
+                    'host_id' : str(uuid.uuid4())}
+                port['host_id'] = host_to_add['host_id']
+                hosts_to_add.append(host_to_add)
+                host_ip_node_lookup_set.add(ip_node_to_find)
+            elif (host is not None):
+                port['host_id'] = host.host_id
+            elif (ip_node_to_find in host_ip_node_lookup_set):
```

Comment:
There is no other possibilities, so this should be just "else"

Reply:
fixed